formatTest/typeCheckedTests/expected_output/jsx.re

```diff
@@ -0,0 +1,368 @@
+  };
+};
+
+let module Foo2 = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let module Text = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let module Exp = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let module Pun = {
+  let createElement intended::intended=? children => {
+    displayName: "test"
+  };
+};
+
+let module Namespace = {
+  let module Foo = {
+    let createElement
+        intended::intended=?
+        children => {
+      displayName: "test"
+    };
+  };
+};
+
+let module LotsOfArguments = {
+  let createElement
+      argument1::argument1=?
+      argument2::argument2=?
+      argument3::argument3=?
+      argument4::argument4=?
+      argument5::argument5=?
+      argument6::argument6=?
+      children => {
+    displayName: "test"
+  };
+};
+
+let div argument1::argument1=? children => {
+  displayName: "test"
+};
+
+let module List1 = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let module List2 = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let module List3 = {
+  let createElement children => {
+    displayName: "test"
+  };
+};
+
+let (/><) a b => a + b;
+
+let (><) a b => a + b;
+
+let (/>) a b => a + b;
+
+let (><\/) a b => a + b;
+
+let tag1 = 5 />< 6;
+
+let tag2 = 5 >< 7;
+
+let tag3 = 5 /> 7;
+
+let tag4 = 5 ><\/ 7;
+
+let b = 2;
+
+let selfClosing = <Foo />;
+
+let selfClosing2 = <Foo a=1 b=true />;
+
+let a = <Foo> <Bar c=(fun a => a + 2) /> </Foo>;
+
+let a3 = <So> <Much> <Nesting /> </Much> </So>;
+
+let a4 =
+  <Sibling>
+    <One test=true foo=b /> <Two foo=b />
+  </Sibling>;
+
+let a5 = <Foo> "testing a string here" </Foo>;
+
+let a6 =
+  <Foo2>
+    <Text> "testing a string here" </Text>
+    <Test yo=1 />
+    <Text> "another string" </Text>
+    <Bar />
+    <Exp> (2 + 4) </Exp>
+  </Foo2>;
+
+let intended = true;
+
+let punning = <Pun intended />;
+
+let namespace = <Namespace.Foo />;
+
+let c = <Foo />;
+
+let d = <Foo />;
+
+let spaceBefore =
+  <So> <Much> <Nesting /> </Much> </So>;
+
+let spaceBefore2 = <So> <Much /> </So>;
+
+let siblingNotSpaced =
+  <So> <Much /> <Much /> </So>;
+
+let jsxInList = [<Foo />];
+
+let jsxInList2 = [<Foo />];
+
+let jsxInListA = [<Foo />];
+
+let jsxInListB = [<Foo />];
+
+let jsxInListC = [<Foo />];
+
+let jsxInListD = [<Foo />];
+
+let jsxInList3 = [<Foo />, <Foo />, <Foo />];
+
+let jsxInList4 = [<Foo />, <Foo />, <Foo />];
+
+let jsxInList5 = [<Foo />, <Foo />];
+
+let jsxInList6 = [<Foo />, <Foo />];
+
+let jsxInList7 = [<Foo />, <Foo />];
+
+let jsxInList8 = [<Foo />, <Foo />];
+
+let testFunc b => b;
+
+let jsxInFnCall = testFunc <Foo />;
+
+let lotsOfArguments =
+  <LotsOfArguments
+     argument1=1
+     argument2=2
+     argument3=3
+     argument4=4
+     argument5=5
+     argument6="test">
+    <Namespace.Foo />
+  </LotsOfArguments>;
+
+let lowerCase = <div argument1=1 />;
+
+let a = <Foo  a>5 </Foo>;
```

Comment:
It's the number `5`. We may want to change how children are represented, or at least *printed*. Right now, the current JavaScript JSX conventions are a special case of what's supported. In `Reason`, the expression `{5}` is the same as `5` (verify for yourself - it's just a sequence expression with one item returned). So we could print it as `<Foo>{5}</Foo>.